### PR TITLE
release: v1.2.0-rc.1

### PR DESCRIPTION
**Pull request by zhaohuabing**
release: v1.2.0-rc.1

### CODE REVIEW

**Review by Xunzhuo — COMMENTED** [review]
Did we mention the eg experimental file mode?

**Comment by zhaohuabing** [comment]
> Did we mention the eg experimental file mode?

Will update the new PRs that got in the last few days.

File: release-notes/v1.2.0-rc.1.yaml [hunk 1]
```diff
@@ -0,0 +1,179 @@
+date: October 23, 2024
```
Comment (zirain) [inline release-notes/v1.2.0-rc.1.yaml:1]: should be `25`?

File: release-notes/v1.2.0-rc.1.yaml [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+
+  - area: xds
+    change: |
+      Bumped go-control-plane to v0.13.1
+
+  - area: cli
+    change: |
+      Added Support for local validations for egctl translate and file provider
+      Fixed egctl experimental translate with error ns
+      Added Support for egctl x collect to collect information from the cluster for debugging
```
Comment (arkodg) [inline release-notes/v1.2.0-rc.1.yaml:179]: can we make `support` in lower case

File: release-notes/v1.2.0-rc.1.yaml [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+      Added startupProbe to all provisioned containers
+
+  - area: xds
+    change: |
+      Bumped go-control-plane to v0.13.1
+
+  - area: cli
+    change: |
+      Added Support for local validations for egctl translate and file provider
+      Fixed egctl experimental translate with error ns
```
Comment (arkodg) [inline release-notes/v1.2.0-rc.1.yaml:178]: not sure what this means, can we add some more clarity like - `fixed an issue in egctl x translate where...`

File: release-notes/v1.2.0-rc.1.yaml [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+
+changes:
+  - area: documentation
+    change: |
+      Fixed capital D in ratelimit example
+      Updated the benchmarking result
+      Added the Envoy Gateway logo into readme
+      Removed redundant commas in the documentation
+      Added the expected release date for v1.2.0
+      Updated the update roadmap page
```
Comment (arkodg) [inline release-notes/v1.2.0-rc.1.yaml:11]:
```suggestion
      Updated the roadmap page
```

File: release-notes/v1.2.0-rc.1.yaml [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+      Added Support for cluster settings for non xRoute-generated backend refs
+      Added Support for socket buffer limit field in ClientTrafficPolicy and BackendTrafficPolicy CRD
+      Added Support for http2 upstream settings in BackendTrafficPolicy CRD
+      Added Support for DNS resolution settings in BackendTrafficPolicy CRD
+      Added Support for direct response in HTTPRouteFilter CRD
+      Added Support for ipv4/ipv6 dual stack in EnvoyProxy and Backend CRD
+
+    breaking-change: |
+      Gateway API GRPCRoute and ReferenceGrant v1alpha2 have been removed.
+      The supportedFeatures field in Gateway API GatewayClass status has changed from being a list of strings to being a list of objects/structs with a name field.
```
Comment (arkodg) [inline release-notes/v1.2.0-rc.1.yaml:77]: this doesnt affect EG, we dont support this field

File: release-notes/v1.2.0-rc.1.yaml [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+
+# Changes that are expected to cause an incompatibility with previous versions, such as deletions or modifications to existing APIs.
+breaking changes: |
+  Gateway API GRPCRoute and ReferenceGrant v1alpha2 have been removed.
+  Please refer to the [Gateway API v1.2.0 documentation](https://github.com/kubernetes-sigs/gateway-api/releases) for more information.
+
+# New features or capabilities added in this release.
+new features: |
+  Added support for Gateway-API v1.2.0
+  Added support for ipv4/ipv6 dual stack for Envoy listeners and Backend resources
```
Comment (arkodg) [inline release-notes/v1.2.0-rc.1.yaml:11]:
```suggestion
  Added support for IPv4/IPv6 Dual Stack for Envoy listeners and BackendRef resources
```

File: release-notes/v1.2.0-rc.1.yaml [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+  Added support for EG standalone(host deployment) mode (experimental)
+  Added support for JWT claims based Authorization in SecurityPolicy CRD
+  Added support for direct response in HTTPRouteFilter CRD
```
Comment (arkodg) [inline release-notes/v1.2.0-rc.1.yaml:14]:
```suggestion
  Added support for Direct Response in HTTPRouteFilter CRD
```

File: release-notes/v1.2.0-rc.1.yaml [hunk 8]
```diff
@@ -0,0 +1,131 @@
+date: October 25, 2024
+
+# Changes that are expected to cause an incompatibility with previous versions, such as deletions or modifications to existing APIs.
+breaking changes: |
+  Gateway API GRPCRoute and ReferenceGrant v1alpha2 have been removed.
+  Please refer to the [Gateway API v1.2.0 documentation](https://github.com/kubernetes-sigs/gateway-api/releases) for more information.
+
```
Comment (arkodg) [inline release-notes/v1.2.0-rc.1.yaml:7]: * Removed default CPU limit of the Envoy Gateway deployment (https://github.com/envoyproxy/gateway/pull/4290)
* Default Envoy Shutdown Settings have changed https://github.com/envoyproxy/gateway/pull/4288 & https://github.com/envoyproxy/gateway/pull/4230
*

Comment (zhaohuabing) [reply release-notes/v1.2.0-rc.1.yaml:7]: Are these considered breaking? Or worth noticing?

File: release-notes/v1.2.0-rc.1.yaml [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+  Added support for non-Kubernetes Backends for TLSRoute
+  Added support for fallback to the Backend API
+  Added support for reloadable EnvoyGateway configuration
+  Added support for adding Labels to the Envoy Service
+  Added support for custom name for ratelimit deployment
+  Added default SecurityContext for EG components
+  Added startupProbe to all provisioned containers
+  Added support for local validations for egctl translate and file provider
+  Added support for egctl x collect to collect information from the cluster for debugging
+
```
Comment (arkodg) [inline release-notes/v1.2.0-rc.1.yaml:62]: * added support for a native prometheus metrics endpoint in the ratelimit server

**Review by arkodg — APPROVED** [review]
LGTM thanks !

**Review by zirain — DISMISSED** [review]

**Comment by zhaohuabing** [comment]
2024-10-25T04:00:56.8174344Z     eg_upgrade.go:91: success to uninstall envoy gateway resources
2024-10-25T04:00:56.8893604Z     eg_upgrade.go:101: failed to  install envoy-gateway: docker.io/envoyproxy/gateway-helm:v1.2.0-rc.1: not found

**Review by arkodg — APPROVED** [review]

**Review by zirain — APPROVED** [review]

**Pull request merged**